lib/rester/client.rb

```diff
@@ -92,28 +95,47 @@ def _init_requester
         ) { |*args| _request(*args) }
 
         @_requester.on_open do
-          logger.error("circuit opened")
+          _log_with_correlation_id(:error, "circuit opened for #{_producer_name}")
         end
 
         @_requester.on_close do
-          logger.info("circuit closed")
+          _log_with_correlation_id(:info, "circuit closed for #{_producer_name}")
         end
       else
         @_requester = proc { |*args| _request(*args) }
       end
     end
 
+    ##
+    # Add a correlation ID to the header and send the request to the adapter
     def _request(verb, path, params)
-      path = _path_with_version(path)
-      _process_response(path, *adapter.request(verb, path, params))
+      _set_default_headers
+      _log_req_res(:request, verb, path)
+      _process_response(verb, path, *adapter.request(verb, path, params))
+    end
+
+    def _set_default_headers
+      adapter.headers(
+        'X-Rester-Correlation-ID' => Rester.correlation_id,
+        'X-Rester-Consumer-Name' => Rester.service_name,
+        'X-Rester-Producer-Name' => _producer_name
+      )
+    end
+
+    def _producer_name
+      @_producer_name ||= "Producer"
```

Comment:
Since we're pinging the service on connect to get the producer name, I feel like if this is nil, we should let it be nil instead of defaulting it to `Producer`.  If someone ever saw a `nil` producer name, it'd be clearer that the producer name wasn't being set, but if it said `Producer` people might go on a wild goose chase to figure out where that was being set.